### PR TITLE
feat/fix: Configuration enhancements

### PR DESCRIPTION
**Pull request by typotter**
## Motivation and Context
Make saving configurations in downstream APIs easier

## Overview of changes
- "grafting" `forServer` field onto the flag config response JSON so that it can be stored with the FCR in a persistent store

### CODE REVIEW

File: src/main/java/cloud/eppo/api/Configuration.java [hunk 1]
```diff
@@ -150,8 +154,11 @@ public Builder(byte[] flagJson, boolean isConfigObfuscated) {
       this.isConfigObfuscated = isConfigObfuscated;
 
       if (flagJson == null || flagJson.length == 0) {
-        throw new RuntimeException(
-            "Null or empty configuration string. Call `Configuration.Empty()` instead");
+        log.warn("Null or empty configuration string. Call `Configuration.Empty()` instead");
```
Comment (aarsilv) [inline src/main/java/cloud/eppo/api/Configuration.java:157]: Ah yeah having this non-fatal is better dev ex.

File: src/main/java/cloud/eppo/api/Configuration.java [hunk 2]
```diff
@@ -230,4 +241,45 @@ public Configuration build() {
...
   }
+
+  /**
+   * Contains the base data required to hydrate a `Configuration` object
+   *
+   * <p>Using `Configuration.writeToSteam` and `Configuration.readFromStream` allows us to avoid
+   * implementing JSON serialization for the `Configuration` class or making all the DTOs `implement
+   * Serializable`
+   */
+  private static class ConfigurationPacket implements Serializable {
```
Comment (aarsilv) [inline src/main/java/cloud/eppo/api/Configuration.java:252]: oh I like this name

File: src/main/java/cloud/eppo/api/Configuration.java [hunk 3]
```diff
@@ -230,4 +241,45 @@ public Configuration build() {
...
+      this.isConfigObfuscated = isConfigObfuscated;
+    }
+  }
+
+  private ConfigurationPacket getConfigurationPacket() {
+    return new ConfigurationPacket(flagConfigJson, banditParamsJson, isConfigObfuscated);
+  }
+
+  public void writeToStream(OutputStream out) throws IOException {
+    ObjectOutputStream oos = new ObjectOutputStream(out);
```
Comment (aarsilv) [inline src/main/java/cloud/eppo/api/Configuration.java:270]: ❗ didn't know about this

However is this the planned use for caching in the filesystem? I think for that we should just save off the original received JSON

**Review by aarsilv — APPROVED** [review]

File: src/main/java/cloud/eppo/api/Configuration.java (hunk 3 above)
Comment (typotter) [reply src/main/java/cloud/eppo/api/Configuration.java]: refactored everything to save the JSON bytes directly to disk. Requires "grafting" the `forServer` field onto the `FlagConfigResponse` json which in turn, requires a deserialization+reserialization before the JSON string is written to disk. This, of course, will be unnecessary when `forServer` is added to the FCR

**Pull request merged**